### ISSUE TITLE
redirect of first page does not work anymore

### ISSUE THREAD

**Issue by fritzmg** [contao/core#8547]
This cannot be tested in the Contao Online Demo directly, since you need to _disable_ **Add the language to the URL**.
1. _Disable_ **Add the language to the URL** in the system settings.
2. Create a website root.
3. Create at least 3 pages: `index`, `page1`, `page2`.
4. Set the type of the `index` page to `forward`.
5. Set its redirect target to `page2` and save.
6. Open the front end with an empty URL (i.e. just `http://example.org/`).

You'll be redirected to the first published regular page, in this case `page1`. In Contao `3.0.0` (?) to `3.5.17` you would have been redirected to the redirect target of the `index` page, in this case `page2`.

I am guessing this happens due to the changes of 32e70a8b290de70f46a8c0c11b585dc66365f5fc (for #8498).

If you _enable_ **Add the language to the URL** everything works as expected, e.g. `http://example.org/` will redirect to `http://example.org/en/page2.html`.

See also [Update auf 3.5.18, Weiterleitungen funktionieren nicht mehr.](https://community.contao.org/de/showthread.php?64414-Update-auf-3-5-18-Weiterleitungen-funktionieren-nicht-mehr/page2)

**Comment by leofeyer** [comment]
See #8498.

**Comment by leofeyer** [comment]
@fritzmg Has this been fixed by #8552, too?

**Comment by olli1770** [comment]
Hi,
I have the same problem as "fritzmg" described above.
I changed the code manually in frontend.php but it doesn`t work anyway.
But the behaviour changed.

Before code-change:
Trying to load logout.html (which is the first real page and not a forward-site) but it endsup in a loop.

After code-change:
Trying to load logout.html but load my 403-page because you can load logout.html only if you are looged in. The loop is gone but the forward-sites are beeing ignored.

**Comment by fritzmg** [comment]
> @fritzmg Has this been fixed by #8552, too?

I'll have to check, but probably not. This code
```php
$objPage = \PageModel::findFirstPublishedRegularByPid($objRootPage->id);

// Redirect if it is not the language fall back page and the alias is "index" (see #8498)
if ($objPage !== null && (!$objRootPage->fallback || $objPage->alias != 'index'))
{
	static::redirect($objPage->getFrontendUrl(), 302);
}
```
finds the first `regular` published page. So if your first page is of type `forward` or `redirect`, it will not be taken into account. I think it would be save to change it to
```php
$objPage = \PageModel::findFirstPublishedByPid($objRootPage->id);
```
This finds any page that is _not_ of type `error_403` or `error_404`.

So if your start page has the alias `index` and it's of the type `forward`, then the page will be loaded as usual and only the page's redirect will be executed.

**Comment by leofeyer** [comment]
Thank you @fritzmg. Fixed in 338bd0763ca2a628e5f80f3afe7c2771c6619a89.
